### PR TITLE
[EXE-1946][EXE-2278] Adding `AiqWeekDiff` and `CollectSet` PushDown Functionality

### PR DESCRIPTION
**Pull request by yannistze**
Adding missing PushDown Support for the following functions:

- `aiq_week_diff`
- `collect_set`

## Test Notes
```
-> % sbt compile
...
[success] Total time: 1 s, completed Feb 9, 2024, 4:19:12 PM
```

```
export SNOWFLAKE_TEST_ACCOUNT="aws"
export SKIP_BIG_DATA_TEST=true
export SPARK_LOCAL_IP=127.0.0.1
export SPARK_CONN_ENV_SFURL=xxxx
export SPARK_CONN_ENV_SFUSER=xxx
export SPARK_CONN_ENV_SFPASSWORD=xxx
export SPARK_CONN_ENV_SFWAREHOUSE=DEMO_WH
export SPARK_CONN_ENV_SFDATABASE=TEST_DB
export SPARK_CONN_ENV_SFSCHEMA=PUBLIC
export SPARK_CONN_ENV_DBTABLE=SPARK_SNOWFLAKE_CI
```

```
sbt "it:testOnly net.snowflake.spark.snowflake.PushdownEnhancement01" >> /Users/yannis.tzemos/Documents/CodeSnips/snow_test_output_1.txt
...
[info] - test pushdown IN() function
[info] - test pushdown INSET() function
[info] - test pushdown INSET() function with timestamps and double
[info] - test pushdown COALESCE()
[info] - pushdown EqualNullSafe: operator <=>
[info] Run completed in 1 minute, 5 seconds.
[info] Total number of tests run: 20
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 20, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 69 s (01:09), completed Feb 9, 2024, 4:20:45 PM
```

```
sbt "it:testOnly net.snowflake.spark.snowflake.PushdownEnhancement02" >> /Users/yannis.tzemos/Documents/CodeSnips/snow_test_output_2.txt
...
[info] - test pushdown number functions: PI() and Round()/Random
[info] - AIQ test pushdown day_start
[info] - AIQ test pushdown day_diff
[info] - AIQ test pushdown week_diff
[info] - AIQ test pushdown collect_set
[info] - AIQ test pushdown instr
[info] - AIQ test pushdown string_to_date
[info] - AIQ test pushdown date_to_string
[info] - AIQ test pushdown md5
[info] - AIQ test pushdown sha1
[info] - AIQ test pushdown sha2
[info] - AIQ test pushdown reverse
[info] - AIQ test pushdown concat_ws
[info] - AIQ test pushdown generate_uuid
[info] - AIQ test pushdown rand
[info] - AIQ test pushdown log
[info] - AIQ test pushdown format_number
[info] - test pushdown functions date_add/date_sub
[info] - test pushdown functions date_add/date_sub on Feb last day in leap and non-leap year
[info] - test pushdown WindowExpression: Rank without PARTITION BY
[info] - test pushdown WindowExpression: Rank with PARTITION BY
[info] - test pushdown WindowExpression: DenseRank without PARTITION BY
[info] - test pushdown WindowExpression: DenseRank with PARTITION BY
[info] - literal Date/Timestamp
[info] - test literal null
[info] - AIQ test approx_count_dist
[info] Run completed in 2 minutes.
[info] Total number of tests run: 28
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 28, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 121 s (02:01), completed Feb 9, 2024, 4:23:18 PM
```

## Deploy
- Merge and `sbt publish`
- Pick up new Connector version in Flame
- Deploy new Clusters

### CODE REVIEW

File: src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala [hunk 1]
```diff
@@ -343,6 +339,157 @@ class PushdownEnhancement02 extends IntegrationSuiteBase {
...
+    val tmpDF = sparkSession.read
+      .format(SNOWFLAKE_SOURCE_NAME)
+      .options(thisConnectorOptionsNoTable)
+      .option("dbtable", test_table_basic)
+      .load()
+
+    val resultDFStr = tmpDF.selectExpr("collect_set(s1)")
+    val resultDFInt = tmpDF.selectExpr("collect_set(s2)")
+
+    // Cannot test the expected result cause the order of
```
Comment (MasterDDT) [inline src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala:458]: Why not collect the items, then sort and compare

Comment (MasterDDT) [reply src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala:458]: Actually if it's deterministic, why isn't it always the same seq?

**Review by MasterDDT — APPROVED** [review]

File: src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala (hunk 1 above)
Comment (yannistze) [reply src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala:458]: So, the order is `non-deterministic` so I can collect sort and compare 👍

Comment (yannistze) [reply src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala:458]: Added the comparison 👍

**Pull request merged**